### PR TITLE
Lessons 01 - 03

### PR DESCRIPTION
**Pull request by teddyboy33**
I think I'm adding lesson01.

### CODE REVIEW

**Comment by teddyboy33** [comment]
While my grid functions work, it felt really inelegant. Did I make it harder on myself? Is there an easier way to go about that?

File: students/philip_korte/lesson03/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    for key, value in donors.items():
+        print(' ',key.title())
+        
+def write_letter(person, donation):
+
+    thank_you = f"""
+    Dear {person.title()},
+    
+    On behalf of all of us here at Warner Bros. we would like to
+    thank you for your generous donation of ${donation:.2f}. Your
```
Comment (majstorpoki) [inline students/philip_korte/lesson03/mailroom.py:30]: Nice formatting!

File: students/philip_korte/lesson03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    print(header)
+    print('-' * l)
+    
+    # sort the dictionary into a new dictionary
+    sorted_donors = dict(sorted(donors.items(), key=lambda e: sum(e[1]), reverse=True))
+
+    # create data summary for each person
+    for key, value in sorted_donors.items():
+        donor_name = key.title()
+        total = sum(value)
```
Comment (majstorpoki) [inline students/philip_korte/lesson03/mailroom.py:56]: Nice usage of sum function!

File: students/philip_korte/lesson03/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        if donation == 'q':
+            # remove new person from donor list before quiting
+            if donors[person] == []:
+                del donors[person]
+            break
+        else:
+            donation = float(donation)
+
+            # add amount to donation history of person
+            donors[person].append(donation)
```
Comment (majstorpoki) [inline students/philip_korte/lesson03/mailroom.py:87]: Tested andworks!

File: students/philip_korte/lesson03/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    header = (f'{"Donor Name":25s} | '
+              f'{"Total Given":>12s} | '
+              f'{"Num Gifts":>12s} | '
+              f'{"Average Gift":>12s} |')
+    l = len(header)
+    print(header)
+    print('-' * l)
+    
+    # sort the dictionary into a new dictionary
+    sorted_donors = dict(sorted(donors.items(), key=lambda e: sum(e[1]), reverse=True))
```
Comment (majstorpoki) [inline students/philip_korte/lesson03/mailroom.py:51]: Nice and elegant!!!!

**Review by majstorpoki — APPROVED** [review]
I am EXTREMELY happy how you solved HW3. I would encourage you to at least start working on Mailroom Pt 2, to keep momentum...

**Pull request closed without merging**